### ISSUE TITLE
CategoricalCrossEntropy Loss runs with wrong tag

### ISSUE THREAD

**Issue by jmaronas** [pytorch/pytorch#16148]
## 🐛 Bug

Putting extreme low negative values as the tag does not raise error

## To Reproduce

       output = torch.randn(5, 2, requires_grad=True)
        target = torch.empty(5, dtype=torch.long).random_(2)
       target[0] = -9223372036854775808

       criterion = nn.CrossEntropyLoss()
        loss = criterion(output, target)
         print(loss)

## Expected behavior

raise Error

## Environment

Last stable version

**Comment by elvisyjlin** [comment]
It's interesting. I looped through negative integers and found that expected error __doesn't__ show up if the value `target[0]` is equal to or less than -100.
